### ISSUE TITLE
Feature Request: Gdrive AppFolder

### ISSUE THREAD

**Issue by pedrosimao** [duplicati/duplicati#2893]
It would be great if we could add Gdrive AppData Folders as our destinations while new backup location.
Gdrive AppData allows us to make data hidden from Gdrive WebUI interface and completely inaccessible to any other app using Gdrive.

An example of an app using this feature:
http://hiddenfolder.flowsoft7.com/
Here we can find the documentation on AppData: https://developers.google.com/drive/v3/web/appdata

Thanks!

**Comment by JonMikelV** [comment]
Does the post here cover what you want?
https://forum.duplicati.com/t/using-google-drive/867/3

**Comment by pedrosimao** [comment]
No. Creating a new oAuth credential will not help me creating an App Folder. Please, take a look at the documentation to understand better what I am suggesting.

**Comment by kenkendk** [comment]
I don't think AppData is appropriate for Duplicati. It works such that if you deauthorize Duplicati from your account, it will automatically delete all stored files (you loose all your backup data).

As @JonMikelV pointed to, Duplicati offers both "Full access (aka Google Docs)" and "Hidden folder (aka Google Drive)". What benefits do you see from using AppData vs. Drive?

**Comment by pedrosimao** [comment]
@kenkendk Well, if you press the "delete" button in a normal folder you will also loose all the data.
Actually, more steps need to be taken, deeper into GDrive UI, to be able to "deauthorize" the app. While deleting is simple as 1,2,3.
I feel that using AppFolder (as an option and not as the default) will add an extra layer of privacy and protection to the data. Privacy because the data will be inaccessible from WebUI and from other apps which have access from API, just as Dupllicatti. And protection because it is a bit more tricky/difficult to "deauthorize" an app than to simplye push the delete button.

**Comment by pedrosimao** [comment]
Furthermore, implementing this "option" is very simple. Few lines of code will do it.

**Comment by ts678** [comment]
>It works such that if you deauthorize Duplicati from your account, it will automatically delete all stored files (you loose all your backup data).

[Hide backup on google drive](https://forum.duplicati.com/t/hide-backup-on-google-drive/17320) independently gave the same objection and more. from one of the few backup programs that do it.

> While deleting a common folder is simple as 1,2,3.

but also obvious that you're deleting your backup. An uninstall of Duplicati leaves the backup data behind. App disconnect kills it.

> What benefits do you see from using AppData vs. Drive?

The major point of that forum request was "It floods your recent files list. If it could utilize App data that would hide it from view."

> Few lines of code will do it.

https://github.com/duplicati/oauth-handler/blob/4e58e68bac8a4bd52e8e96b8d186951dde33467a/settings.py#L242-L261

is probably the spot. I'm still not convinced that the issues of being so fragile (it's supposed to be configuration data, and in that sense it makes more sense for it to go away when the app goes away) and inaccessible (see cited issues, etc.) are worth benefits. General unwillingness of backup programs to do this (feel free to research it, as I did) should serve as a cautionary note on this...
